lib/chess/core/cell.rb

```diff
@@ -0,0 +1,55 @@
+require 'colorize'
+require 'colorized_string'
+
+require_relative './chess'
+
+module Chess
+  module Core
+    class Cell
+      attr_reader :occupant
+
+      Algebraic = Struct.new(:column, :row) do
+        def to_s
+          "#{column}#{row}"
+        end
+      end
+
+      def initialize(name, fill_color)
+        @occupied = false
+        @occupant = nil
+        @fill_color = fill_color
+        @name = name
+      end
+
+      def occupy_with(piece)
+        @occupied = true
+        @occupant = piece
+      end
+
+      def release
+        @occupied = false
+        @occupant = nil
+      end
+
+      def occupied?
+        @occupied
+      end
```

Comment:
No need to have `@occupied` as an instance attribute. It can be a method that depends on `@occupant`
```suggestion
      def occupied?
        !!@occupant
      end
```

Reply:
done